gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -410,6 +461,134 @@ ChannelCredentials createMtlsChannelCredentials() throws IOException, GeneralSec
     return null;
   }
 
+  /**
+   * Create the S2A-Secured Channel credentials. Load the API using reflection. Once the S2A API is
+   * stable in gRPC-Java, all callers of this method can simply use the S2A APIs directly and this
+   * method can be deleted.
+   *
+   * @param s2aAddress the address of the S2A server used to secure the connection.
+   * @param s2aChannelCredentials the credentials to be used when connecting to the S2A.
+   * @return {@code ChannelCredentials} instance.
+   */
+  ChannelCredentials buildS2AChannelCredentials(
+      String s2aAddress, ChannelCredentials s2aChannelCredentials) {
+    try {
+      // Load the S2A API.
+      Class<?> s2aChannelCreds = Class.forName("io.grpc.s2a.S2AChannelCredentials");
+      Class<?> s2aChannelCredsBuilder = Class.forName("io.grpc.s2a.S2AChannelCredentials$Builder");
+
+      // Load and invoke the S2A API methods.
+      Class<?>[] partypes = new Class[2];
+      partypes[0] = String.class;
+      partypes[1] = ChannelCredentials.class;
+      Method newBuilder = s2aChannelCreds.getMethod("newBuilder", partypes);
+      Object arglist[] = new Object[2];
+      arglist[0] = s2aAddress;
+      arglist[1] = s2aChannelCredentials;
+      Object retObjBuilder = newBuilder.invoke(null, arglist);
+      Method build = s2aChannelCredsBuilder.getMethod("build", null);
+      Object retObjCreds = build.invoke(retObjBuilder, null);
+      return (ChannelCredentials) retObjCreds;
+    } catch (Throwable t) {
+      LOG.log(Level.INFO, "S2A APIs cannot be used: ", t.getMessage());
```

Comment:
I think this should be an WARNING level message. If this is unable to be created, it would fallback to the normal flow? If so, can that also be added to the log statement?

Reply:
> If this is unable to be created, it would fallback to the normal flow?
Yes, it would fallback to normal flow (TLS without S2A).

> I think this should be an WARNING level message...If so, can that also be added to the log statement?
Done.

https://github.com/googleapis/sdk-platform-java/pull/3548/commits/b80729413fd6002220874159159d05704e499dac